contracts/staking/ValidatorManager.sol

```diff
@@ -269,57 +269,56 @@ abstract contract ValidatorManager is
      *  Note that this function can be used for successful validation periods that have been explicitly
      * ended by calling {initializeEndValidation} or for validation periods that never began on the P-Chain due to the
      * {registrationExpiry} being reached.
-     * @param setWeightMessageType Whether or not the message type is a SetValidatorWeight message, or a
-     * SubnetValidatorRegistration message (with valid set to false). Both message types are valid for ending
-     * a validation period.
      */
-    function completeEndValidation(uint32 messageIndex, bool setWeightMessageType) external {
+    function completeEndValidation(uint32 messageIndex) external {
         ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
 
         // Get the Warp message.
         (WarpMessage memory warpMessage, bool valid) =
             WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
-        require(valid, "ValidatorManager: Invalid warp message");
-
-        bytes32 validationID;
-        if (setWeightMessageType) {
-            uint64 weight;
-            (validationID,, weight) =
-                ValidatorMessages.unpackSetSubnetValidatorWeightMessage(warpMessage.payload);
-            require(weight == 0, "ValidatorManager: Weight must be 0");
-        } else {
-            bool validRegistration;
-            (validationID, validRegistration) =
-                ValidatorMessages.unpackSubnetValidatorRegistrationMessage(warpMessage.payload);
-            require(!validRegistration, "ValidatorManager: Registration still valid");
-        }
+        require(valid, "ValidatorManager: invalid warp message");
```

Comment:
nit: Since we have `validRegistration` below, can we rename `valid` here to `validWarpMessage` or something similar?

Reply:
Going to leave this as is because it'll be encapsulated in the aforementioned [P-Chain helper](https://github.com/ava-labs/teleporter/pull/503#discussion_r1735284703) shortly